### DISCUSSION TITLE
Receiver Action about Packet loss

### DISCUSSION THREAD

**Discussion by JaesukLim** [ANLAB-KAIST/KENSv3#230] [Project Q&A 2. Transport Layer (TCP)] [answered]
To handle TestTransfer_Connect_Recv_Symmetric (unreliable), I am confusing the discard action
At the test case, the server is sending seq=65025 to seq=115713 while the seq=87553 is corrupted. Then after they get duplicate ack's of 87553, they still send with 116225. And after all the data sent, they retransmit the data only 87553 to 112641.

It seems the client keeps the data even if there was corrupted packet. How can I understand the phenomenon? I am struggling to get the point of the solution build.

**Comment by Pusnow** [comment]
Would you upload or send your pcap file? cs341.pj@kaist.ac.kr

**Reply by Pusnow** [reply]
> It seems the client keeps the data even if there was corrupted packet.

And yes, the solution only drops corrupted packets.

**Reply by JaesukLim** [reply]
I sent the pcap with the email!

**Reply by Pusnow** [reply, marked as answer]
You may be asking why the sender sends next data before retransmitting dropped packet.

At the sender's view, before being acknowledged by duplicated acks, it receives successful acks, freeing receiver window. So, the sender transmits remaining data to the receiver.

And after receiving the duplicated acks, it retransmits packets until getting notified.

BTW, you don't need to follow the exact behavior of the solution as lone as you pass all test cases.
